backend/eventloop/shared_resource_loop/sharedresourceloop_test.go

```diff
@@ -947,10 +938,115 @@ var _ = Describe("SharedResourceEventLoop Test", func() {
+			Expect(err).ToNot(HaveOccurred())
+
+			err = dbq.CreateGitopsEngineInstance(ctx, &gitopsEngineInstance)
+			Expect(err).ToNot(HaveOccurred())
+
+			cr := &managedgitopsv1alpha1.GitOpsDeploymentRepositoryCredential{
+				ObjectMeta: metav1.ObjectMeta{
+					Name:      "test-gitopsdeploymenrepositorycredential",
+					Namespace: gitopsEngineInstance.Namespace_name,
+					UID:       uuid.NewUUID(),
+				},
+				Spec: managedgitopsv1alpha1.GitOpsDeploymentRepositoryCredentialSpec{
+					Repository: "https://fakegithub.com/test/test-repository",
+					Secret:     "test-secret",
+				}}
+
+			err = k8sClient.Create(ctx, cr)
+			Expect(err).ToNot(HaveOccurred())
+
+			// Create new Secret
+			secret := &v1.Secret{
+				ObjectMeta: metav1.ObjectMeta{
+					Name:      "test-secret",
+					Namespace: gitopsEngineInstance.Namespace_name,
+				},
+				Data: map[string][]byte{
+					"username": []byte("test-username"),
+					"password": []byte("test-password"),
+				},
+			}
+			err = k8sClient.Create(ctx, secret)
+			Expect(err).ToNot(HaveOccurred())
+
+			defer dbq.CloseDatabase()
+			var repositoryCredentialCRNamespace v1.Namespace
+			repositoryCredentialCRNamespace.Name = gitopsEngineInstance.Namespace_name
+			repositoryCredentialCRNamespace.UID = types.UID(gitopsEngineInstance.Namespace_uid)
+			var k8sClientFactory SRLK8sClientFactory
+			dbRepoCred, err := internalProcessMessage_ReconcileRepositoryCredential(ctx, cr.Name, repositoryCredentialCRNamespace, k8sClient, k8sClientFactory, dbq, false, l)
+			Expect(err).ToNot(HaveOccurred())
+			Expect(dbRepoCred).ToNot(BeNil())
+
+			sharedResourceEventLoopRepoCred, err := sharedResourceEventLoop.ReconcileRepositoryCredential(ctx, k8sClient, repositoryCredentialCRNamespace, cr.Name, MockSRLK8sClientFactory{fakeClient: k8sClient})
+			Expect(sharedResourceEventLoopRepoCred.RepositoryCredentialsID).To(Equal(dbRepoCred.RepositoryCredentialsID))
+			Expect(err).ToNot(HaveOccurred())
+			// To be used by AfterEach to clean up the resources created by test
 			resourcesToBeDeleted = testResources{
-				AppProjectRepository: getappProjectRepositoryDB,
+				Gitopsengineinstance_id: gitopsEngineInstance.Gitopsengineinstance_id,
+				EngineCluster_id:        gitopsEngineInstance.EngineCluster_id,
+				Clustercredentials_id: []string{
+					clusterCredentials.Clustercredentials_cred_id,
+				},
 			}
+
+		})
+
+		It("Should test ReconcileRepositoryCredential when RepositoryCredential CR is nil", func() {
+			sharedResourceEventLoop := &SharedResourceEventLoop{inputChannel: make(chan sharedResourceLoopMessage)}
+
+			go internalSharedResourceEventLoop(sharedResourceEventLoop.inputChannel)
+
+			sharedResourceLoop, err := sharedResourceEventLoop.ReconcileRepositoryCredential(ctx, k8sClient, *namespace, "test-name", MockSRLK8sClientFactory{fakeClient: k8sClient})
+			Expect(err).ToNot(HaveOccurred())
+			Expect(sharedResourceLoop).To(BeNil())
+
+			resourcesToBeDeleted = testResources{}
```

Comment:
Added empty resourcesToBeDeleted here as DeferCleanUp function is giving this error 
```
[FAILED] Expected
      <int>: 0
  to equal
      <int>: 1
```
May be this is a bug, or please let me know if no resource needs to be deleted we should pass empty resourcesToBeDeleted or not?